jsonschema-module-swagger-2/src/test/java/com/github/victools/jsonschema/module/swagger2/IntegrationTest.java

```diff
@@ -0,0 +1,137 @@
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import org.junit.runner.RunWith;
+
+/**
+ * Integration test of this module being used in a real SchemaGenerator instance.
+ */
+@RunWith(JUnitParamsRunner.class)
+public class IntegrationTest {
+
+    public Object[] parametersForTestIntegration() {
+        return new Object[][]{
+            {TestClass.class},
+            {Foo.class}
+        };
+    }
+
+    @Test
+    @Parameters
+    public void testIntegration(Class<?> rawTargetType) throws Exception {
+        Swagger2Module module = new Swagger2Module();
+        SchemaGeneratorConfigBuilder configBuilder = new SchemaGeneratorConfigBuilder(SchemaVersion.DRAFT_2019_09, OptionPreset.PLAIN_JSON)
+                .with(module);
+        SchemaGenerator generator = new SchemaGenerator(configBuilder.build());
+        JsonNode result = generator.generateSchema(rawTargetType);
+
+        String rawJsonSchema = result.toString();
+        JSONAssert.assertEquals('\n' + rawJsonSchema + '\n',
+                loadResource("integration-test-result-" + rawTargetType.getSimpleName() + ".json"), rawJsonSchema,
+                JSONCompareMode.STRICT);
+    }
+
+    private static String loadResource(String resourcePath) throws IOException {
+        StringBuilder stringBuilder = new StringBuilder();
+        try (InputStream inputStream = IntegrationTest.class.getResourceAsStream(resourcePath);
+                Scanner scanner = new Scanner(inputStream, StandardCharsets.UTF_8.name())) {
+            while (scanner.hasNext()) {
+                stringBuilder.append(scanner.nextLine()).append('\n');
+            }
+        }
+        String fileAsString = stringBuilder.toString();
+        return fileAsString;
+
+    }
+
+    @Schema(title = "test title", description = "test description")
+    static class TestClass {
+
+        @Schema(hidden = true)
+        public Object hiddenField;
+
+        @Schema(name = "fieldWithOverriddenName")
+        public List<Boolean> originalFieldName;
+
+        @Schema(description = "field description", nullable = true, allowableValues = {"A", "B", "C", "D"})
+        public String fieldWithDescriptionAndAllowableValues;
+
+        @Schema(minimum = "15", maximum = "20")
+        public int fieldWithInclusiveNumericRange;
+
+        @Schema(minimum = "14", maximum = "21", exclusiveMinimum = true, exclusiveMaximum = true)
+        public int fieldWithExclusiveNumericRange;
+    }
+
+    interface IReference {
+
+        String getName();
+    }
+
+    class Reference<T> implements IReference {
+
+        private String name;
+
+        @Override
+        public String getName() {
+            return this.name;
+        }
+    }
+
+    class Person {
+
+    }
+
+    @Schema(description = "the foo's person")
+    class PersonReference extends Reference<Person> {
```

Comment:
In my real use-case, this would also be an interface, i.e. `interface PersonReference extends IReference`. That breaks the properties resolution because unfortunately the getter in the interface won't be detected as such.